application/helpers/common_helper.php

```diff
@@ -3999,6 +3999,26 @@ function questionAttributes($returnByName=false)
         "help"=>gT('Use button group or radio list'),
         "caption"=>gT('Display type'));
 
+        /**
+         * New event to allow plugin to add own question attribute (settings)
+         * Using $event->append('questionAttributes', $questionAttributes);
+         * $questionAttributes=[
```

Comment:
Sure, I just wanted you to consider creating a class to contain the data and behaviour instead of using arrays and loose functions. See this: https://github.com/LimeSurvey/LimeSurvey/blob/master/application/libraries/MenuObjects/MenuItem.php